app/utils/security.js

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {Client4} from 'mattermost-redux/client';
+import CookieManager from 'react-native-cookies';
+import urlParse from 'url-parse';
+
+export function setCSRFFromCookie(url) {
+    CookieManager.get(urlParse(url).origin, false).then((res) => {
+        const token = res.MMCSRF;
+        if (token) {
+            Client4.setCSRF(res?.MMCSRF?.value || res?.MMCSRF);
```

Comment:
Why do we need to check both `res.MMCSRF` or `res.MMCSRF.value` for the token? Shouldn't it always be just one or the other?

Also, we could simplify this by removing most of the `?.`s since we already know that `res.MMCSRF` is defined here because it's the same thing as `token` which we know is truthy

Reply:
For some particular reason @enahum and me couldnt figure out, on Android the value is directly in the `res.MMCSRF`, while on iOS `res.MMCSRF` is an object where value contains the actual cookie. I'll simply the second part of it though.

Reply:
Ah, strange. I guess that's just the library being weird